src/vaults/PoolManager.sol

```diff
@@ -137,6 +137,7 @@ contract PoolManager is
     function registerAsset(uint16 centrifugeId, address asset, uint256 tokenId)
         external
         payable
+        protected
```

Comment:
Don't see a clear risk, but since there is external calls, seems worth adding the re-entrancy modifier.